platform-modules/ekstep-framework/src/main/java/org/ekstep/framework/mgr/impl/FrameworkManagerImpl.java

```diff
@@ -84,61 +69,44 @@ public Response createFramework(Map<String, Object> request, String channelId) t
 
 	}
 
-	/*
-	 * Read framework by Id
-	 * 
-	 * @param graphId
-	 * 
+	/**
+	 * Read Framework By Id
 	 * @param frameworkId
-	 * 
-	 */
-	// TODO : Uncomment this method
-	/*
-	 * @Override public Response readFramework(String frameworkId) throws Exception
-	 * { return read(frameworkId, FRAMEWORK_OBJECT_TYPE,
-	 * FrameworkEnum.framework.name());; }
+	 * @param returnCategories
+	 * @return Response
+	 * @throws Exception
 	 */
-
-	// TODO : Delete this method and uncomment above method
 	@SuppressWarnings("unchecked")
 	@Override
 	public Response readFramework(String frameworkId, List<String> returnCategories) throws Exception {
 		Response response = new Response();
-		if (Platform.config.hasPath("framework.es.sync")) {
-			if (Platform.config.getBoolean("framework.es.sync")) {
-				Map<String, Object> responseMap = new HashMap<>();
-				List<Object> searchResult = searchFramework(frameworkId);
-				if (null != searchResult && !searchResult.isEmpty()) {
-					Map<String, Object> framework = (Map<String, Object>) searchResult.get(0);
-					if (null != framework.get("fw_hierarchy")) {
-						Map<String, Object> hierarchy = mapper.readValue((String) framework.get("fw_hierarchy"),
-								Map.class);
-						responseMap = framework;
-						if (null != hierarchy && !hierarchy.isEmpty()) {
-							List<Map<String, Object>> categories = (List<Map<String, Object>>) hierarchy
-									.get("categories");
-							if (categories != null) {
-								if (returnCategories != null && !returnCategories.isEmpty()) {
-									responseMap.put("categories",
-											categories.stream().filter(p -> returnCategories.contains(p.get("code")))
-													.collect(Collectors.toList()));
-									removeAssociations(responseMap, returnCategories);
-
-								} else {
-									responseMap.put("categories", categories);
-								}
-							}
+		Map<String, Object> responseMap = new HashMap<>();
+		Response getHierarchyResp = getFrameworkHierarchy(frameworkId);
+		Map<String, Object> framework = (Map<String, Object>) getHierarchyResp.get("framework");
+
+		if (MapUtils.isNotEmpty(framework)) {
+			if (null != framework.get("fw_hierarchy")) {
```

Comment:
@krgauraw Do you feel, still we should use the same logic.. We can store and fetch directly complete framework hierarchy

Reply:
we are storing complete framework data not only hierarchy. So it is required as of now